components/common/src/main/java/com/hotels/styx/common/content/FlowControllingPublisher.java

```diff
@@ -0,0 +1,53 @@
+/*
+  Copyright (C) 2013-2020 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.common.content;
+
+import com.hotels.styx.api.Buffer;
+import io.netty.channel.EventLoop;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+/**
+ * A publisher to wrap the FlowControllingHttpContentProducer FSM and perform subscription operations via a Netty EventLoop.
+ */
+public final class FlowControllingPublisher implements Publisher<Buffer> {
```

Comment:
Frankly, the class name is quite meaningless. Normally I try to avoid debating names, but in this case it is justified to rename this class to `NettyByteStream`. Reasoning below...

We use this class to adapt `FlowControllingHttpContentProducer` to `ByteStream`. This uses uses Netty to source the byte stream content from a TCP socket.

In this context, I'd suggest you to rename this to `NettyByteStreamPublisher`. But because `ByteStream` already is a publisher, therefore `NettyByteStream` is sufficient.